pkg/assets/builder.go

```diff
@@ -21,43 +21,57 @@ import (
 	"fmt"
 	"net/url"
 	"os"
+	"path"
 	"strings"
 
+	"github.com/golang/glog"
 	"k8s.io/kops/pkg/apis/kops"
 	"k8s.io/kops/pkg/featureflag"
 	"k8s.io/kops/pkg/kubemanifest"
+	"k8s.io/kops/util/pkg/hashing"
+	"k8s.io/kops/util/pkg/vfs"
 )
 
 // RewriteManifests controls whether we rewrite manifests
 // Because manifest rewriting converts everything to and from YAML, we normalize everything by doing so
 var RewriteManifests = featureflag.New("RewriteManifests", featureflag.Bool(true))
 
-// AssetBuilder discovers and remaps assets
+// AssetBuilder discovers and remaps assets.
 type AssetBuilder struct {
 	ContainerAssets []*ContainerAsset
 	FileAssets      []*FileAsset
 	AssetsLocation  *kops.Assets
+	// TODO we'd like to use cloudup.Phase here, but that introduces a go cyclic dependency
+	Phase string
 }
 
+// ContainerAsset models a container's location.
 type ContainerAsset struct {
-	// DockerImage will be the name of the docker image we should run, if this is a docker image
+	// DockerImage will be the name of the docker image we should run.
+	// This is used to copy a file to a ContainerRegistry.
```

Comment:
The "this is used to copy a file...." is on the CanonicalLocation for FileAsset but the cluster-location-field for ContainerAsset.

Reply:
More context please

Reply:
Looks like you got it in another commit